tesseract_common/include/tesseract_common/class_loader.h

```diff
@@ -90,10 +112,31 @@ struct ClassLoader
 };
 }  // namespace tesseract_common
 
-#define TESSERACT_ADD_PLUGIN(DERIVED_CLASS, ALIAS)                                                                     \
+// clang-format off
+#define TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, SECTION)                                                  \
   extern "C" BOOST_SYMBOL_EXPORT DERIVED_CLASS ALIAS;                                                                  \
+  BOOST_DLL_SECTION(SECTION, read) BOOST_DLL_SELECTANY                                                                 \
   DERIVED_CLASS ALIAS;
 
+#define TESSERACT_ADD_PLUGIN(DERIVED_CLASS, ALIAS)                                                                     \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, boostdll)
```

Comment:
What do you anticipate using this for? It might be confusing to user creating a plugin if they find this macro before finding a macro for the specific plugin type they actually care about. If it doesn't have a direct use case now (other than testing) you might consider removing it

Reply:
Not sure just following the boost pattern of available macros.

Reply:
I don't think this is needed then. The idea is to be able to search for plugins of specific types; the value comes from the other macro definitions that are specific to `tesseract` classes not this generic one

Reply:
I would like to leave this since I plan to break this out into its own package separated from Tesseract to allow use by other packages. I think this could be used by some of the other ros agnostic packages.